front-end/src/components/CreateEditPostModal.tsx

```diff
@@ -53,52 +98,89 @@ export default function CreatePostComponent(props: Props){
         }
       }
 
-      axios.post(process.env.REACT_APP_API_URL + "/api/author/" + props.loggedInUser.authorId + "/posts/", data, config)
-      .then(res => {
-        if (res.status >= 400) {
-          setShowError(true)
-        } else if (res.status === 201) {
-          console.log('WORKED!!!')
+      if(isCreate){
+        axios.post(process.env.REACT_APP_API_URL + "/api/author/" + props.loggedInUser.authorId + "/posts/", data, config)
+          .then(res => {
+            handleRes(res)
+          }).catch(error => {
+            setShowError(true)
+          })
+      }
+      else if(props.editFields !== undefined){
+          axios.put(process.env.REACT_APP_API_URL + "/api/author/" + props.loggedInUser.authorId + "/posts/" + props.editFields.id + "/", data, config)
```

Comment:
editing posts should do a POST request and PUT should be only used when a author-specified post ID is present

Reply:
This might explain the server issue that I was having. Let me play around with it for a bit